From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mauro Rossi <issor.oruam@gmail.com>
Date: Sat, 28 Sep 2019 18:13:54 +0200
Subject: [PATCH 12/54] x86/intel: force tsc to be reliable on Baytrail (v2)

(v1) original Chih-Wei Huang commit 3e4251a

(v2) change due to commits:
     bba10c5 ("x86/cpu: Use constant definitions for CPU models")
     0cc5359 ("x86/cpu: Update init data for new Airmont CPU model")
---
 arch/x86/kernel/cpu/intel.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/x86/kernel/cpu/intel.c b/arch/x86/kernel/cpu/intel.c
index 816fdbec795a..7d07e8d1e2b5 100644
--- a/arch/x86/kernel/cpu/intel.c
+++ b/arch/x86/kernel/cpu/intel.c
@@ -256,6 +256,8 @@ static void early_init_intel(struct cpuinfo_x86 *c)
 	/* Penwell and Cloverview have the TSC which doesn't sleep on S3 */
 	if (c->x86 == 6) {
 		switch (c->x86_model) {
+		case INTEL_FAM6_ATOM_SILVERMONT:
+			set_cpu_cap(c, X86_FEATURE_TSC_RELIABLE);
 		case INTEL_FAM6_ATOM_SALTWELL_MID:
 		case INTEL_FAM6_ATOM_SALTWELL_TABLET:
 		case INTEL_FAM6_ATOM_SILVERMONT_MID:
